pkg/sources/forward/applier/sourcetransformer.go

```diff
@@ -25,13 +25,13 @@ import (
 // SourceTransformApplier applies the source transform on the read message and gives back a new message. Any UserError will be retried here, while
 // InternalErr can be returned and could be retried by the callee.
 type SourceTransformApplier interface {
-	ApplyTransform(ctx context.Context, message *isb.ReadMessage) ([]*isb.WriteMessage, error)
+	ApplyTransform(ctx context.Context, messages []*isb.ReadMessage) ([]isb.ReadWriteMessagePair, error)
```

Comment:
Currently, we run the below func in goroutines and call `ApplyTransform` within it: 
```go
concurrentApplyTransformer(ctx context.Context, readMessagePair <-chan *isb.ReadWriteMessagePair)
```
It retrieves message from the `readMessagePair` channel, apply transformer, and store the `WriteMessages` in the same element.

With bidrectional streaming, the concurrency is moved to the server side. It might be difficult to have a function signature that takes a message and returns the output since we are not sure of the result order from the stream. Instead, we send all messages, wait for all output and aggregate the result in the new implementation.